provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/contract/ContractApiHttp.java

```diff
@@ -111,6 +119,54 @@ public static ContractApiHttp createFromConnectionString(
         return new ContractApiHttp(provisioningConnectionString);
     }
 
+    /**
+     * Create a new instance of the ContractApiHttp with a custom {@link TokenCredential} to allow for finer grain control
+     * of authentication tokens used in the underlying connection.
+     *
+     * @param credential The custom {@link TokenCredential} that will provide authentication tokens to
+     *                                    this library when they are needed. The provided tokens must be Json Web Tokens.
+     */
+    public ContractApiHttp(String hostName, TokenCredential credential)
+    {
+        if (Tools.isNullOrEmpty(hostName))
+        {
+            /* SRS_HTTP_DEVICE_REGISTRATION_CLIENT_21_007: [The constructor shall throw IllegalArgumentException if the provided hostName is null or empty.] */
```

Comment:
All of these "SRS_..." comments in the source code and the test code are no longer required in this repo, so feel free to just delete them